Source/CarthageKit/BackTrackingResolver.swift

```diff
@@ -139,9 +139,12 @@ public final class BackTrackingResolver: ResolverProtocol {
 				}
 			}
 		}
-
-		// By definition result is not nil at this point, so it's ok to force unwrap
-		return result!
+		
+		// By definition result is not nil at this point (while loop only breaks when result is not nil)
+		guard let finalResult = result else {
+			preconditionFailure("Expected result to not be nil")
```

Comment:
What I mean here is `preconditionFailure` will crash carthage and (I think) won't give any good reason to the user.

I would rather prefer to throw a CarhageError

Reply:
This is like an assertion, as you can see from the program flow this should never happen